From 2f9ebfc205f6aa006ef13f9a542126ea4d0c5273 Mon Sep 17 00:00:00 2001
From: Sami Shalayel <sami.shalayel@qt.io>
Date: Wed, 1 Oct 2025 13:54:28 +0200
Subject: [PATCH 190/239] tst_qqmlecmascript: add data test for date parsing

Add data test for date parsing, move 2 testcases from checkDateTime to
checkDateTimeParsing() and remove the now unused qml files. This
prepares for the fix of QTBUG-100377 that will need more such data
parsing tests.

Pick-to: 6.8
Task-number: QTBUG-100377
Change-Id: Id99920402d2aa141bc4d13c88a472f0b6ad3c65b
Reviewed-by: Fabian Kosmale <fabian.kosmale@qt.io>
(cherry picked from commit 98e68b996a4dedc3fcdb4ffdf1088982b595351c)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../data/checkDateTime-nonstandardFormat2.qml | 10 ------
 .../data/checkDateTimeParsing.qml             |  9 ++++++
 .../qml/qqmlecmascript/tst_qqmlecmascript.cpp | 31 +++++++++++++++++--
 3 files changed, 37 insertions(+), 13 deletions(-)
 delete mode 100644 tests/auto/qml/qqmlecmascript/data/checkDateTime-nonstandardFormat2.qml
 create mode 100644 tests/auto/qml/qqmlecmascript/data/checkDateTimeParsing.qml

diff --git a/tests/auto/qml/qqmlecmascript/data/checkDateTime-nonstandardFormat2.qml b/tests/auto/qml/qqmlecmascript/data/checkDateTime-nonstandardFormat2.qml
deleted file mode 100644
index 3fd3c05024..0000000000
--- a/tests/auto/qml/qqmlecmascript/data/checkDateTime-nonstandardFormat2.qml
+++ /dev/null
@@ -1,10 +0,0 @@
-import Qt.test 1.0
-import QtQml 2.15
-
-
-MyTypeObject {
-    Component.onCompleted: {
-        dateTimeProperty = new Date("Sun, 25 Mar 2018 11:10:49 GMT")
-        boolProperty = !Number.isNaN(dateTimeProperty)
-    }
-}
diff --git a/tests/auto/qml/qqmlecmascript/data/checkDateTimeParsing.qml b/tests/auto/qml/qqmlecmascript/data/checkDateTimeParsing.qml
new file mode 100644
index 0000000000..980956b98c
--- /dev/null
+++ b/tests/auto/qml/qqmlecmascript/data/checkDateTimeParsing.qml
@@ -0,0 +1,9 @@
+import QtQuick
+
+
+Item {
+    property string myString
+    property date myDateTime: new Date(myString)
+    property real myTime: myDateTime.getTime()
+    property bool isValid: !Number.isNaN(myDateTime)
+}
\ No newline at end of file
diff --git a/tests/auto/qml/qqmlecmascript/tst_qqmlecmascript.cpp b/tests/auto/qml/qqmlecmascript/tst_qqmlecmascript.cpp
index 54994ff5e7..cd3dd4cdd9 100644
--- a/tests/auto/qml/qqmlecmascript/tst_qqmlecmascript.cpp
+++ b/tests/auto/qml/qqmlecmascript/tst_qqmlecmascript.cpp
@@ -73,6 +73,8 @@ private slots:
     void checkDate();
     void checkDateTime_data();
     void checkDateTime();
+    void checkDateTimeParsing_data();
+    void checkDateTimeParsing();
     void idShortcutInvalidates();
     void boolPropertiesEvaluateAsBool();
     void methods();
@@ -741,9 +743,6 @@ void tst_qqmlecmascript::checkDateTime_data()
     QTest::newRow("nonstandard-format")
         << testFileUrl("checkDateTime-nonstandardFormat.qml")
         << QDateTime::fromString("1991-08-25 20:57:08 GMT+0000", "yyyy-MM-dd hh:mm:ss t");
-    QTest::newRow("nonstandard-format2")
-        << testFileUrl("checkDateTime-nonstandardFormat2.qml")
-        << QDateTime::fromString("Sun, 25 Mar 2018 11:10:49 GMT", "ddd, d MMM yyyy hh:mm:ss t");
 }
 
 void tst_qqmlecmascript::checkDateTime()
@@ -760,6 +759,32 @@ void tst_qqmlecmascript::checkDateTime()
     QVERIFY(object->boolProperty());
 }
 
+void tst_qqmlecmascript::checkDateTimeParsing_data()
+{
+    QTest::addColumn<QString>("string");
+    QTest::addColumn<QDateTime>("when");
+
+    QTest::newRow("rfc2822-with-timezone-name")
+            << u"Sun, 25 Mar 2018 11:10:49 GMT"_s
+            << QDateTime(QDate(2018, 3, 25), QTime(11, 10, 49), QTimeZone::utc());
+}
+
+void tst_qqmlecmascript::checkDateTimeParsing()
+{
+    QFETCH(QString, string);
+    QFETCH(const QDateTime, when);
+
+    QQmlEngine e;
+    QQmlComponent component(&e, testFileUrl("checkDateTimeParsing.qml"));
+    QScopedPointer<QObject> obj(component.create());
+    QVERIFY2(obj, qPrintable(component.errorString()));
+
+    obj->setProperty("myString", string);
+    QCOMPARE(obj->property("myDateTime").toDateTime(), when);
+    QCOMPARE(obj->property("myTime").toLongLong(), when.toMSecsSinceEpoch());
+    QCOMPARE(obj->property("isValid").toBool(), when.isValid());
+}
+
 void tst_qqmlecmascript::idShortcutInvalidates()
 {
     QQmlEngine engine;
-- 
2.51.2

